### PR TITLE
Display tiles + players + food count + rocks count + eggs

### LINKED ISSUE #100: Display tiles + players + food count + rocks count + eggs
### 1. Display all tiles
Open a window, display all the tiles with the correct positioning

### 2. Display all players + eggs
For each tile, display the players + eggs on this tile (each having the color of their team)

### 3. Display Food count
For each tile, display the amount of food that is on the tile

### 4. Display Rocks count
For each tile, display the total amount of rocks that are on this tile, no matter the type

### PR DESCRIPTION
**Pull request by Rafouzou**
### 1. Display all tiles
Open a window, display all the tiles with the correct positioning

### 2. Display all players + eggs
For each tile, display the players + eggs on this tile (each having the color of their team)

### 3. Display Food count
For each tile, display the amount of food that is on the tile

### 4. Display Rocks count
For each tile, display the total amount of rocks that are on this tile, no matter the type

Closes #100

### CODE REVIEW

File: gui/src/data_parsing/get_active_player_data.cpp [hunk 1]
```diff
@@ -34,13 +34,13 @@ void Data::startIncantationFromString(std::string t_str)
...
         throw(GuiError("Gui: Data: startIncantationFromString(" + t_str + "): Error parsing command"));
-    } else if (!(current_player = find_player(m_teams, id))) {
+    } else if (!(current_player = findPlayer(m_teams, id))) {
         throw(GuiError("Gui: Data: startIncantationFromString(" + t_str + "): No player with this ID was found"));
     }
     players.push_back(current_player);
     while (!line_ss.eof()) {
         line_ss >> next_id;
-        if (!(current_player = find_player(m_teams, next_id))) {
+        if (!(current_player = findPlayer(m_teams, next_id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_active_player_data.cpp:43]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_active_player_data.cpp [hunk 2]
```diff
@@ -34,13 +34,13 @@ void Data::startIncantationFromString(std::string t_str)
         throw(GuiError("Gui: Data: startIncantationFromString(" + t_str + "): Command '" + command + "' unknown"));
     } else if (line_ss.fail()) {
         throw(GuiError("Gui: Data: startIncantationFromString(" + t_str + "): Error parsing command"));
-    } else if (!(current_player = find_player(m_teams, id))) {
+    } else if (!(current_player = findPlayer(m_teams, id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_active_player_data.cpp:37]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_active_player_data.cpp [hunk 3]
```diff
@@ -112,7 +112,7 @@ void Data::startEggLayingFromString(std::string t_str)
         throw(GuiError("Gui: Data: startEggLayingFromString(" + t_str + "): Command '" + command + "' unknown"));
     } else if (line_ss.fail()) {
         throw(GuiError("Gui: Data: startEggLayingFromString(" + t_str + "): Error parsing command"));
-    } else if (!(player = find_player(m_teams, id))) {
+    } else if (!(player = findPlayer(m_teams, id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_active_player_data.cpp:115]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_active_player_data.cpp [hunk 4]
```diff
@@ -145,15 +145,15 @@ void Data::endEggLayingFromString(std::string t_str)
         throw(GuiError("Gui: Data: endEggLayingFromString(" + t_str + "): Command '" + command + "' unknown"));
     } else if (line_ss.fail()) {
         throw(GuiError("Gui: Data: endEggLayingFromString(" + t_str + "): Error parsing command"));
-    } else if (!(team = find_player_team(m_teams, player_id))) {
+    } else if (!(team = findPlayerTeam(m_teams, player_id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_active_player_data.cpp:148]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_active_player_data.cpp [hunk 5]
```diff
@@ -145,15 +145,15 @@ void Data::endEggLayingFromString(std::string t_str)
...
         throw(GuiError("Gui: Data: endEggLayingFromString(" + t_str + "): No player with this player ID was found"));
-    } else if ((egg = find_egg(m_teams, egg_id))) {
+    } else if ((egg = findEgg(m_teams, egg_id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_active_player_data.cpp:150]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_general_data.cpp [hunk 6]
```diff
@@ -129,7 +129,7 @@ void Data::endGameFromString(std::string t_str)
         line_ss >> team_name_bit;
         team_name = team_name + " " + team_name_bit;
     }
-    if (!(team = find_team(m_teams, team_name))) {
+    if (!(team = findTeam(m_teams, team_name))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_general_data.cpp:132]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_map_data.cpp [hunk 7]
```diff
@@ -53,7 +53,7 @@ void Data::dropResourceFromString(std::string t_str)
         throw(GuiError("Gui: Data: dropResourceFromString(" + t_str + "): Command '" + command + "' unknown"));
     } else if (line_ss.fail()) {
         throw(GuiError("Gui: Data: dropResourceFromString(" + t_str + "): Error parsing command"));
-    } else if (!(player = find_player(m_teams, player_id))) {
+    } else if (!(player = findPlayer(m_teams, player_id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_map_data.cpp:56]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_map_data.cpp [hunk 8]
```diff
@@ -112,7 +112,7 @@ void Data::collectResourceFromString(std::string t_str)
         throw(GuiError("Gui: Data: collectResourceFromString(" + t_str + "): Command '" + command + "' unknown"));
     } else if (line_ss.fail()) {
         throw(GuiError("Gui: Data: collectResourceFromString(" + t_str + "): Error parsing command"));
-    } else if (!(player = find_player(m_teams, player_id))) {
+    } else if (!(player = findPlayer(m_teams, player_id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_map_data.cpp:115]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_passive_player_data.cpp [hunk 9]
```diff
@@ -45,14 +45,14 @@ void Data::addPlayerFromString(std::string t_str)
     if (!(player_direction > 0 && player_direction < 5)) {
         throw(GuiError("Gui: Data: addPlayerFromString(" + t_str + "): Player direction should be an integer contained between 1 and 4"));
     }
-    if (!(team_by_name = find_team(m_teams, player_team_name))) {
+    if (!(team_by_name = findTeam(m_teams, player_team_name))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_passive_player_data.cpp:48]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_passive_player_data.cpp [hunk 10]
```diff
@@ -45,14 +45,14 @@ void Data::addPlayerFromString(std::string t_str)
...
         m_teams.push_back(std::make_shared<Team>(player_team_name));
         team_by_name = m_teams[m_teams.size() - 1];
     }
-    if ((team_by_player_id = find_player_team(m_teams, player_id)) && (team_by_name != team_by_player_id)) {
+    if ((team_by_player_id = findPlayerTeam(m_teams, player_id)) && (team_by_name != team_by_player_id)) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_passive_player_data.cpp:52]: it is forbidden to compare and add in a variable at the same time

File: gui/src/data_parsing/get_passive_player_data.cpp [hunk 11]
```diff
@@ -45,14 +45,14 @@ void Data::addPlayerFromString(std::string t_str)
...
         throw(GuiError("Gui: Data: addPlayerFromString(" + t_str + "): Another player with the same ID was found in a different team"));
     }
-    if (!(player = find_player_in_team(team_by_name, player_id))) {
+    if (!(player = findPlayerInTeam(team_by_name, player_id))) {
```
Comment (lbarreteau) [inline gui/src/data_parsing/get_passive_player_data.cpp:55]: it is forbidden to compare and add in a variable at the same time

File: gui/src/game.cpp [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+    m_window.clear();
+    drawTiles();
+    m_window.display();
+}
+
+/**
+ * @brief Updates screen size
+ * @details Calculates ans sets new screen size based on terrain size
+*/
+void Game::updateSize(void)
```
Comment (lbarreteau) [inline gui/src/game.cpp:78]: not explicit => updateScreenSize or something like this

File: gui/src/game.cpp [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+    std::deque<std::shared_ptr<Player>> players;
+    std::deque<sf::Color> colors;
+    for (auto team : m_data->m_teams) {
+        for (auto player : team->m_players) {
+            if (player->getX() == t_tile->getX() && player->getY() == t_tile->getY()) {
+                players.push_back(player);
+                colors.push_back(m_team_colors[team_index]);
+            }
+        }
+        team_index += 1;
```
Comment (lbarreteau) [inline gui/src/game.cpp:150]: add the increment to the for

File: gui/src/game.cpp [hunk 14]
```diff
@@ -0,0 +1,250 @@
...
+    }
+    max_player_index = std::max(players.size() - 1, (std::size_t)1);
+    for (auto player : players) {
+        m_player_sprite.setPosition(
+            player->getX() * (TILE_SIZE + TILE_SPACING) + SCREEN_MARGIN + TILE_SIZE - PLAYER_SIZE - ((TILE_SIZE - PLAYER_SIZE) * player_index / max_player_index),
+            player->getY() * (TILE_SIZE + TILE_SPACING) + SCREEN_MARGIN + TILE_SIZE - PLAYER_SIZE);
+        m_player_sprite.setColor(colors[player_index]);
+        m_window.draw(m_player_sprite);
+        player_index += 1;
```
Comment (lbarreteau) [inline gui/src/game.cpp:159]: Same

**Review by lbarreteau — CHANGES_REQUESTED** [review]

**Review by HugoN1 — CHANGES_REQUESTED** [review]

File: gui/Makefile [hunk 15]
```diff
@@ -28,13 +27,14 @@ SRC 		= src/client/IClient/Client.cpp						\
...
-			  src/serverMessage.cpp
+			  src/serverMessage.cpp								\
+			  src/game/game.cpp
 
 OBJ     	= $(SRC:%.cpp=%.o) $(MAIN:%.cpp=%.o)
 
 CXX			= g++
 
-CXXFLAGS	= -Wall -Wextra -Werror -Iincludes -I../includes -Isrc -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio
+CXXFLAGS	= -Wall -Wextra -Werror -Iincludes -I../includes -Isrc -Isrc/game -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio
```
Comment (lbarreteau) [inline gui/Makefile:37]: Please remove -Isrc/game

File: gui/src/game/game.cpp [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+ * @param tile Tile to display
+*/
+void Game::drawTilePlayers(std::shared_ptr<Tile> t_tile)
+{
+    unsigned int team_index = 0;
+    unsigned int player_index = 0;
+    unsigned int max_player_index = 0;
+    std::deque<std::shared_ptr<Player>> players;
+    std::deque<sf::Color> colors;
+    for (auto team : m_data->m_teams) {
```
Comment (lbarreteau) [inline gui/src/game/game.cpp:132]: space between for and variables

File: gui/src/game/game.cpp [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+        for (auto player : team->m_players) {
+            if (player->getX() == t_tile->getX() && player->getY() == t_tile->getY()) {
+                players.push_back(player);
+                colors.push_back(m_team_colors[team_index]);
+            }
+        }
+        team_index += 1;
```
Comment (lbarreteau) [inline gui/src/game/game.cpp:139]: team_index += 1 in the for like has previously said

File: gui/src/game/game.cpp [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+    }
+    max_player_index = std::max(players.size() - 1, (std::size_t)1);
+    for (auto player : players) {
+        m_player_sprite.setPosition(
+            player->getX() * (TILE_SIZE + TILE_SPACING) + SCREEN_MARGIN + TILE_SIZE - PLAYER_SIZE - ((TILE_SIZE - PLAYER_SIZE) * player_index / max_player_index),
+            player->getY() * (TILE_SIZE + TILE_SPACING) + SCREEN_MARGIN + TILE_SIZE - PLAYER_SIZE);
+        m_player_sprite.setColor(colors[player_index]);
+        m_window.draw(m_player_sprite);
+        player_index += 1;
```
Comment (lbarreteau) [inline gui/src/game/game.cpp:148]: player_index += 1 in the for like has previously said

File: gui/src/game/game.cpp [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+        float saturation = 1.0f;
+        float value = 1.0f;
+        unsigned int color_zone = static_cast<unsigned int>(hue / 60) % 6;
+        float f = hue / 60 - color_zone;
+        float p = value * (1 - saturation);
+        float q = value * (1 - f * saturation);
+        float t = value * (1 - (1 - f) * saturation);
+        float r = 0;
+        float g = 0;
+        float b = 0;
```
Comment (lbarreteau) [inline gui/src/game/game.cpp:225]: image

declare variable in loop, is very very forbiden

**Review by lbarreteau — CHANGES_REQUESTED** [review]

**Review by lbarreteau — APPROVED** [review]

File: gui/sprites/diamond.png
Comment (AndreasLETANTER) [inline gui/sprites/diamond.png:1]: why double diamond png ?

**Review by AndreasLETANTER — APPROVED** [review]

**Review by HugoN1 — CHANGES_REQUESTED** [review]

**Review by HugoN1 — APPROVED** [review]

**Pull request merged**